### PR TITLE
CONSOLE-2371: [i18n] Externalize olm modal strings

### PR DESCRIPTION
**Pull request by TheRealJon**
Externalize strings in `frontend/packages/operator-lifecycle-manager/src/components/modals`

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/modals/delete-catalog-source-modal.tsx [hunk 1]
```diff
@@ -35,27 +37,30 @@ const DeleteCatalogSourceModal: React.FC<DeleteCatalogSourceModalProps> = ({
   return (
     <form onSubmit={submit} name="form" className="modal-content ">
       <ModalTitle>
-        <YellowExclamationTriangleIcon className="co-icon-space-r" /> Delete {kind.label}?
+        <YellowExclamationTriangleIcon className="co-icon-space-r" />{' '}
+        {t('olm~Delete {{kind}}?', { kind: kind.label })}
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/delete-catalog-source-modal.tsx:41]:
```suggestion
        {t('olm~Delete CatalogSource?')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/modals/delete-catalog-source-modal.tsx [hunk 2]
```diff
@@ -35,27 +37,30 @@ const DeleteCatalogSourceModal: React.FC<DeleteCatalogSourceModalProps> = ({
...
       </ModalTitle>
       <ModalBody>
         <p>
-          By deleting a catlog source, any operator that has been installed from this source will no
-          longer receive updates.
+          {t(
+            'olm~By deleting a catlog source, any operator that has been installed from this source will no longer receive updates.',
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/delete-catalog-source-modal.tsx:46]:
```suggestion
            'olm~By deleting a CatlogSource, any operator that has been installed from this source will no longer receive updates.',
```

File: frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.tsx [hunk 3]
```diff
@@ -17,91 +16,90 @@ import {
...
-  render() {
-    return (
-      <form onSubmit={this.submit.bind(this)} name="form" className="modal-content">
-        <ModalTitle className="modal-header">Change Update Approval Strategy</ModalTitle>
-        <ModalBody>
-          <div className="co-m-form-row">
-            <p>What strategy is used for approving updates?</p>
+  return (
+    <form onSubmit={submit} name="form" className="modal-content">
+      <ModalTitle className="modal-header">{t('olm~Change Update Approval Strategy')}</ModalTitle>
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.tsx:56]:
```suggestion
      <ModalTitle className="modal-header">{t('olm~Change update approval strategy')}</ModalTitle>
```

File: frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.tsx [hunk 4]
```diff
@@ -17,91 +16,90 @@ import {
...
+        <div className="co-m-form-row">
+          <p>{t('olm~What strategy is used for approving updates?')}</p>
+        </div>
+        <div className="co-m-form-row row">
+          <div className="col-sm-12">
+            <RadioInput
+              onChange={(e) => setSelectedApprovalStrategy(e.target.value)}
+              value={InstallPlanApproval.Automatic}
+              checked={selectedApprovalStrategy === InstallPlanApproval.Automatic}
+              title={t(`olm~${InstallPlanApproval.Automatic}`)}
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.tsx:67]: I'd do it this way when possible.

```suggestion
              title={t('olm~Automatic')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.tsx [hunk 5]
```diff
@@ -17,91 +16,90 @@ import {
...
-                  </p>
-                </div>
-              </RadioInput>
-            </div>
+          <div className="col-sm-12">
+            <RadioInput
+              onChange={(e) => setSelectedApprovalStrategy(e.target.value)}
+              value={InstallPlanApproval.Manual}
+              checked={selectedApprovalStrategy === InstallPlanApproval.Manual}
+              title={t(`olm~${InstallPlanApproval.Manual}`)}
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.tsx:82]:
```suggestion
              title={t('olm~Manual')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-preview-modal.tsx [hunk 6]
```diff
@@ -10,29 +10,34 @@ import { ActionGroup, Button } from '@patternfly/react-core';
...
-      </ActionGroup>
-    </ModalFooter>
-  </div>
-);
+const InstallPlanPreview: React.FC<InstallPlanPreviewModalProps> = ({ cancel, stepResource }) => {
+  const { t } = useTranslation();
+  return (
+    <div className="modal-content">
+      <ModalTitle>
+        {t('olm~{{kind}} Preview', { kind: InstallPlanModel.label })}{' '}
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-preview-modal.tsx:21]:
```suggestion
        {t('olm~InstallPlan preview')}{' '}
```

File: frontend/packages/operator-lifecycle-manager/src/components/modals/subscription-channel-modal.tsx [hunk 7]
```diff
@@ -1,82 +1,88 @@
...
-          submitDisabled={this.state.selectedChannel === getSelectedChannel(this.props)}
-        />
-      </form>
-    );
-  }
-}
+  return (
+    <form onSubmit={submit} name="form" className="modal-content">
+      <ModalTitle className="modal-header">
+        {t('olm~Change Subscription Update Channel')}
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/subscription-channel-modal.tsx:49]:
```suggestion
        {t('olm~Change Subscription update channel')}
``

File: frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx [hunk 8]
```diff
@@ -1,110 +1,113 @@
...
-        <YellowExclamationTriangleIcon className="co-icon-space-r" /> Uninstall Operator?
+        <YellowExclamationTriangleIcon className="co-icon-space-r" /> {t('olm~Uninstall Operator?')}
       </ModalTitle>
       <ModalBody>
-        This will remove operator <strong>{name}</strong> from {context}. Removing the operator will
-        not remove any of its custom resource definitions or managed resources. If your operator has
-        deployed applications on the cluster or configured off-cluster resources, these will
-        continue to run and need to be cleaned up manually.
+        <Trans t={t} ns="olm">
+          This will remove operator <strong>{{ name }}</strong> from{' '}
```
Comment (spadgett) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx:84]: Do we capitalize "Operator" elsewhere? We should try to be consistent whatever we choose.

Comment (TheRealJon) [reply frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx:84]: Yeah I think we do, I'll try to catch these wherever I see them

**Comment by TheRealJon** [comment]
@spadgett I've addressed your feedback and tests are passing. I think this one is ready to merge.

File: frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.spec.tsx [hunk 9]
```diff
@@ -45,7 +52,7 @@ describe(InstallPlanApprovalModal.name, () => {
   it('renders a modal form', () => {
     expect(wrapper.find('form').props().name).toEqual('form');
     expect(wrapper.find(ModalTitle).exists()).toBe(true);
-    expect(wrapper.find(ModalSubmitFooter).props().submitText).toEqual('Save');
+    expect(wrapper.find(ModalSubmitFooter).props().submitText).toEqual('olm~Save');
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.spec.tsx:55]:
```suggestion
    expect(wrapper.find(ModalSubmitFooter).props().submitText).toEqual('public~Save');
```

File: frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.tsx [hunk 10]
```diff
@@ -17,91 +16,90 @@ import {
...
-    );
-  }
-}
+        </div>
+      </ModalBody>
+      <ModalSubmitFooter
+        inProgress={inProgress}
+        errorMessage={errorMessage}
+        cancel={cancel}
+        submitText={t('olm~Save')}
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/installplan-approval-modal.tsx:97]:
```suggestion
        submitText={t('public~Save')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/modals/subscription-channel-modal.spec.tsx [hunk 11]
```diff
@@ -76,25 +83,25 @@ describe(SubscriptionChannelModal.name, () => {
   it('renders a modal form', () => {
     expect(wrapper.find('form').props().name).toEqual('form');
     expect(wrapper.find(ModalTitle).exists()).toBe(true);
-    expect(wrapper.find(ModalSubmitFooter).props().submitText).toEqual('Save');
+    expect(wrapper.find(ModalSubmitFooter).props().submitText).toEqual('olm~Save');
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/subscription-channel-modal.spec.tsx:86]:
```suggestion
    expect(wrapper.find(ModalSubmitFooter).props().submitText).toEqual('public~Save');
```

File: frontend/packages/operator-lifecycle-manager/src/components/modals/subscription-channel-modal.tsx [hunk 12]
```diff
@@ -1,82 +1,88 @@
...
+              />
+            </div>
+          ))}
+        </div>
+      </ModalBody>
+      <ModalSubmitFooter
+        inProgress={inProgress}
+        errorMessage={errorMessage}
+        cancel={cancel}
+        submitText={t('olm~Save')}
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/subscription-channel-modal.tsx:80]:
```suggestion
        submitText={t('public~Save')}
```

File: frontend/packages/operator-lifecycle-manager/src/components/modals/update-strategy-modal.tsx [hunk 13]
```diff
@@ -73,12 +76,12 @@ export const UpdateStrategyModal = withHandlePromise((props: UpdateStrategyModal
       <ModalSubmitFooter
         errorMessage={errorMessage}
         inProgress={inProgress}
-        submitText="Save"
+        submitText={t('olm~Save')}
```
Comment (rhamilto) [inline frontend/packages/operator-lifecycle-manager/src/components/modals/update-strategy-modal.tsx:79]:
```suggestion
        submitText={t('public~Save')}
```

**Review by rhamilto — COMMENTED** [review]
Save is already in the `public` namespace, so probably don't need to add it to `olm`, too.

**Comment by TheRealJon** [comment]
@rhamilto Addressed your comments

**Pull request merged**